modules/core/src/bitgo.ts

```diff
@@ -70,59 +83,54 @@ superagent.Request.prototype.end = function(cb) {
 
 // Handle HTTP errors appropriately, returning the result body, or a named
 // field from the body, if the optionalField parameter is provided.
-superagent.Request.prototype.result = function(optionalField) {
+(superagent as any).Request.prototype.result = function(optionalField?: string) {
   return this.then(handleResponseResult(optionalField), handleResponseError);
 };
 
-const handleResponseResult = function(optionalField: any = false) {
-  return function(res) {
+function handleResponseResult(optionalField?: string): (res: superagent.Response) => any {
+  return function(res: superagent.Response) {
     if (_.isNumber(res.status) && res.status >= 200 && res.status < 300) {
       return optionalField ? res.body[optionalField] : res.body;
     }
     throw errFromResponse(res);
   };
-};
+}
 
-const errFromResponse = function(res) {
+function errFromResponse(res: superagent.Response): Error {
   const errString = createResponseErrorString(res);
   const err: any = new Error(errString);
 
   err.status = res.status;
   if (res.body) {
     err.result = res.body;
   }
-  if (_.has(res.headers, 'x-auth-required') && (res.headers['x-auth-required'] === 'true')) {
+  if (_.has(res.header, 'x-auth-required') && (res.header['x-auth-required'] === 'true')) {
```

Comment:
why is this singular now?

Reply:
res.headers does not exist, this was always broken as far as I can tell:

From superagent docs:

> Response header fields
> 
> The res.header contains an object of parsed header fields, lowercasing field names much like node does. For example res.header['content-length'].

Reply:
woah that's wild

Reply:
thats some 2015 Davenport code